### ISSUE TITLE
Unable to re-establish connection to central after failure (ver. 0.10.9)

### ISSUE THREAD

**Issue by aoganesi** [glowroot/glowroot#434]
Hi,

We ran into an issue today that our central server was not available at the time when one of the agents (running under tomcat) was restarted. The system keeps reporting reconnect failures and no data is being transmitted since even though central is available and working now. Is this a known issue?

``
2018-07-24 02:00:56.561 WARN  o.g.agent.central.CentralConnection - unable to establish connection with the central collector https://glowroot.ligolab.com:443 (will keep trying...): null
``

**Comment by trask** [comment]
Hi @aoganesi, just want to confirm, the agent looks like it might be pointing to the central collector UI port? Or do you have the gRPC port running on 443 and UI port running on some other port? Thanks.

**Comment by trask** [comment]
@aoganesi Thanks for reporting this. This is not a known issue. Unfortunately the warn message `null` doesn't give much of a clue why this happened. I have updated that warn message to display the exception name as well, hopefully that will give more of a clue in the future. For now, if you continue to see this (or are able to repro?) can you set the agent log level to debug on `org.glowroot.agent.central.CentralConnection` so we can get the full stack trace for this error? You can set this log level by adding a file named `glowroot.logback.xml` in your agent directory with the contents:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE configuration>
<configuration>
  <appender name="CONSOLE" class="org.glowroot.agent.shaded.ch.qos.logback.core.ConsoleAppender">
    <encoder>
      <pattern>%d{yyyy-MM-dd HH:mm:ss.SSS} %-5level %logger{36} - %msg%n</pattern>
    </encoder>
  </appender>
  <appender name="FILE" class="org.glowroot.agent.shaded.ch.qos.logback.core.rolling.RollingFileAppender">
    <rollingPolicy class="org.glowroot.agent.shaded.ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
      <fileNamePattern>${glowroot.log.dir}/glowroot.%d{yyyy-MM-dd}.log</fileNamePattern>
      <maxHistory>7</maxHistory>
    </rollingPolicy>
    <encoder>
      <!-- including date even though it is in filename to make it easier on log aggregators -->
      <pattern>%d{yyyy-MM-dd HH:mm:ss.SSS} %-5level %logger{36} - %msg%n</pattern>
    </encoder>
  </appender>
  <logger name="org.glowroot.agent.shaded" level="warn" />
  <logger name="org.glowroot.agent.embedded.shaded" level="warn" />
  <logger name="org.glowroot.agent.shaded.io.netty.handler.codec.http2.Http2ConnectionHandler"
    level="off" />
  <logger name="org.glowroot.agent.central.CentralConnection" level="debug" />
  <root level="info">
    <appender-ref ref="CONSOLE" />
    <appender-ref ref="FILE" />
  </root>
  <logger name="audit" level="off" />
</configuration>
```

(this will override the default logback configuration found under `glowroot.jar/glowroot.logback.xml`)

**Comment by trask** [comment]
Hi @aoganesi, I think there's a good chance the issue you ran into is fixed by a796242a05fd29f0cebdc1941eaca32a22eec35f in the latest snapshot release [glowroot-agent-0.11.0-SNAPSHOT-dist.zip](https://oss.sonatype.org/service/local/artifact/maven/redirect?r=snapshots&g=org.glowroot&a=glowroot-agent&c=dist&e=zip&v=0.11.0-SNAPSHOT). Previously the agent was only resolving the collector host name at startup, which could lead to problems.

Please re-open if you see this again after upgrading (hopefully v0.11.0 will be out soon).

**Comment by aoganesi** [comment]
Thank you! Will update as soon as it's out. Greatly appreciate the attention!
